src/server/frontend_xwayland/xwayland_clipboard_provider.cpp

```diff
@@ -240,84 +345,28 @@ void mf::XWaylandClipboardProvider::send_data(
         if (!current_source)
         {
             log_warning("failed to send clipboard data to X11 client: no source");
-            send_selection_notify(time, requester, XCB_ATOM_NONE, connection.CLIPBOARD, target);
+            send_selection_notify(*connection, time, requester, XCB_ATOM_NONE, connection->CLIPBOARD, target);
             return;
         }
         current_source->initiate_send(mime_type, in_fd);
-        in_fd = {}; // important to release ownership of the fd, as we are about to block on it closing
-    }
-
-    size_t const data_size = increment_chunk_size;
-    std::unique_ptr<uint8_t[]> const data{new uint8_t[data_size]};
-
-    size_t len;
-    try
-    {
-        len = read_repeatedly(out_fd, data.get(), data_size);
-    }
-    catch (ReadError const& err)
-    {
-        log_warning("failed to send clipboard data to X11 client: failed to read from fd: %s", err.what());
-        send_selection_notify(time, requester, XCB_ATOM_NONE, connection.CLIPBOARD, target);
-        return;
-    }
-
-    if (len < data_size)
-    {
-        // Call the XCB function directly instead of using connection->set_property() because target is variable
-        xcb_change_property(
-            connection,
-            XCB_PROP_MODE_REPLACE,
-            requester,
-            property,
-            target,
-            8, // format
-            len,
-            data.get());
-        send_selection_notify(time, requester, property, connection.CLIPBOARD, target);
-    }
-    else
-    {
-        log_warning("failed to send clipboard data to X11 client: incremental transfers not yet implemented");
-        send_selection_notify(time, requester, XCB_ATOM_NONE, connection.CLIPBOARD, target);
-        // TODO: remember to flush the connection before reading
-        return;
+        in_fd = {}; // release ownership of the fd here
     }
```

Comment:
I realize this is pre-existing, but why does `in_fd` have a scope that outlasts its use? Fixing that would avoid this ugly reset. (OTOH why does it even exist?)